### ISSUE TITLE
Update vectorized setting with new name and defaults

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/docs#5141]
tl;dr: `sql.defaults.experimental_vectorize` is now
`sql.defaults.vectorize`, and it has some new defaults.

Via @yuzefovich comment on #4953:

A quick note: we have just merged a PR ([cockroachdb/cockroach#38777][PR]) that renamed `experimental_vectorize` to just `vectorize`. It also added a new possible mode of vectorization, and now all modes are:

- `off` - the vectorized engine is disabled, all queries go through the row execution engine.
- `auto` (which is now the default choice) - all queries consisting only of streaming operators (a streaming operator is such that doesn't require any buffering) are executed through the vectorized engine whereas all others are run through the row execution engine.
- `experimental_on` - all queries that are supported in the vectorized engine (both streaming and non-streaming) are run through it. It is "experimental" because we still don't have disk spilling (which means that large queries can get out of memory error and crash the node).
- `experimental_always` - absolutely all queries are forced to run through the vectorized engine. If the engine doesn't support the query, it errors out. The only exception is SET queries so that vectorize session variable can be changed.> 
[PR]: https://github.com/cockroachdb/cockroach/pull/38777

**Comment by yuzefovich** [comment]
Another quick note: https://github.com/cockroachdb/cockroach/pull/39285 removed `experimental_always` from the list of valid options for cluster setting `sql.defaults.vectorize`; however, it is still possible to set session variable `vectorize` to `experimental_always`.
